File: patat/code/gcds.c
#include <stdlib.h>

int gcd(int a, int b) { return !b ? a : gcd(b, a%b); }

int gcds (int capacity, size_t n, int weights[n]) {
	int i, gcd_all = capacity;
	for (i = 0; i < n; i++) {
		gcd_all = gcd (gcd_all, weights[i]);
	}
	return gcd_all;
}

File: patat/code/gcds.java
class GCDS {

public int gcds (int capacity, int[] weights) {
	int gcd_all = capacity;
	for (int weight : weights) {
		gcd_all = gcd (gcd_all, weight);
	}
	return gcd_all;
}

public int gcd(int a, int b) { return b==0 ? a : gcd(b, a%b); }

public static void main () {
	System.out.println(gcds(CAP, WS));
}
}